src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -96,44 +155,63 @@ int main(int argc, char* argv[])
 
         // read data (iNumOfFrames might be updated!)
         phAudioFile->readData(ppfAudioData, iNumFrames);
-
+		//if (pCCombFilter) 
+		{
+			pCCombFilter->process(ppfAudioData, ppfAudioDataOut, iNumFrames, stFileSpec.iNumChannels);
+			if (phAudioFile2)
+				phAudioFile2->writeData(ppfAudioDataOut, iNumFrames);
+		}
         cout << "\r" << "reading and writing";
 
         // write
         for (int i = 0; i < iNumFrames; i++)
         {
+			
             for (int c = 0; c < stFileSpec.iNumChannels; c++)
             {
                 hOutputFile << ppfAudioData[c][i] << "\t";
+				hOutputFile << ppfAudioDataOut[c][i] << "\t";
             }
             hOutputFile << endl;
         }
     }
 
     cout << "\nreading/writing done in: \t" << (clock() - time) * 1.F / CLOCKS_PER_SEC << " seconds." << endl;
-
+	phAudioFile2->closeFile();
     //////////////////////////////////////////////////////////////////////////////
     // clean-up (close files and free memory)
     CAudioFileIf::destroy(phAudioFile);
     hOutputFile.close();
 
-    for (int i = 0; i < stFileSpec.iNumChannels; i++)
-        delete[] ppfAudioData[i];
+	for (int i = 0; i < stFileSpec.iNumChannels; i++) {
+		delete[] ppfAudioData[i];
+		delete[] ppfAudioDataOut[i];
+	}
+	delete[] ppfAudioDataOut;
     delete[] ppfAudioData;
     ppfAudioData = 0;
-
+	ppfAudioDataOut = 0;
     // all done
     return 0;
 
 }
 
 
-void     showClInfo()
+void showClInfo()
 {
     cout << "MUSI6106 Assignment Executable" << endl;
     cout << "(c) 2014-2022 by Alexander Lerch" << endl;
     cout  << endl;
-
     return;
 }
+void errorMsg(char *msg)
+{
+	printf("%s", msg);
+	printf("\r\n");
+	printf("the Program syntax: ./MUSI6106Exec inputfile.wav -f/i n\r\n");
+	printf("-t f means use FIR, -t i means use IIR.\r\n");
+	printf("n specify the max delay in millsecond.\r\n,some examples is below\r\n");
+	printf("./MUSI6106Exec la.wav -f 100\r\n");
+	printf("./MUSI6106Exec la.wav -i 150\r\n");
+}
```

Comment:
This is very informative, I like the fact that you've added this to the executable.